File: 2025/D1/U1/Main.java
package D1.U1;

import java.io.File;
import java.io.FileNotFoundException;
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        File file = new File("./inputs/1.input");
        int password = 0;
        int current = 50;

        try (Scanner reader = new Scanner(file)) {
            while (reader.hasNextLine()) {
                String data = reader.nextLine();
                if (data.charAt(0) == 'L') {
                    current = (current - Integer.parseInt(data.substring(1))) % 100;
                } else if (data.charAt(0) == 'R') {
                    current = (current + Integer.parseInt(data.substring(1))) % 100;
                }
                if (current == 0) {
                    password++;
                }
            }
        } catch (FileNotFoundException e) {
            System.out.println("Error");
            e.printStackTrace();
        }
        System.out.println(password);
    }
}

File: 2025/D1/U1/Main.py
password = 0

current = 50

with open('2025/inputs/1.input', 'r') as f:
    lines = f.read().split('\n')

for line in lines:
    if line[0] == 'L':
        current = (current - int(line[1:])) % 100
    else:
        current = (current + int(line[1:])) % 100
    
    if current == 0:
        password += 1

print(password)